### PR TITLE
feature/big-query-argument-setter

### PR DESCRIPTION
**Pull request by ardian4**
Added: BigQueryArgumentSetter
Added: BigQueryArgumentSetterConfig
Added: Widget BigQueryArgumentSetter-action.json
Added: Docs BigQueryArgumentSetter-action.md

### CODE REVIEW

**Comment by ardian4** [comment]
@googlebot I signed it!

File: src/test/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        .setTable(VALID_TABLE)
+        .setArgumentSelectionConditions(VALID_ARGUMENT_SELECTION_CONDITIONS)
+        .setArgumentsColumns(VALID_ARGUMENT_COLUMN);
+  }
+
+  private static void validateConfigValidationFail(BigQueryArgumentSetterConfig config) {
+    MockFailureCollector collector = new MockFailureCollector();
+    try {
+      config.validateProperties(collector);
+      Assert.assertEquals(1, collector.getValidationFailures().size());
```
Comment (CuriousVini) [inline src/test/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfigTest.java:41]: we should assert specific failure type not just size.

File: src/test/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package io.cdap.plugin.gcp.bigquery.action;
```
Comment (CuriousVini) [inline src/test/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfigTest.java:1]: missing copyright. 
Are you able to build the plugins using - `mvn clean package -DskipTests`?

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfig.java [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+  private void validateArgumentsSelectionConditions(
+      String argumentSelectionConditions, FailureCollector collector) {
+    if (containsMacro(NAME_ARGUMENT_SELECTION_CONDITIONS)) {
+      return;
+    }
+
+    if (Strings.isNullOrEmpty(argumentSelectionConditions)) {
+      collector.addFailure(
+          "Arguments Selection Conditions is empty!",
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfig.java:160]: nit: `empty.`
Please add period to end of failure message and corrective action

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetter.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+              queryResults.getTotalRows()));
+    }
+
+    Schema schema = queryResults.getSchema();
+    FieldValueList row = queryResults.iterateAll().iterator().next();
+
+    for (int i = 0; i < schema.getFields().size(); i++) {
+      Field field = schema.getFields().get(i);
+      String name = field.getName();
+      String stringValue = row.get(name).getStringValue();
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetter.java:101]: we should make sure that schema only has primitive types as per the javadoc of `getStringValue `

File: src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfig.java [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+      Map<String, String> argumentConditionMap, Map<String, Field> argumentConditionFields) {
+    if (argumentConditionMap.size() == argumentConditionFields.size()) {
+      return;
+    }
+    String nonExistingColumnNames =
+        argumentConditionMap.keySet().stream()
+            .filter(columnName -> !argumentConditionFields.containsKey(columnName))
+            .collect(Collectors.joining(" ,"));
+    throw new RuntimeException(
+        String.format("Those Columns Name \" %s \"do not exits in Table", nonExistingColumnNames));
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfig.java:220]: `Columns: %s do not exist in table.`
What is the corrective action that user needs to take in this case?

**Review by CuriousVini — COMMENTED** [review]
Please add copyright to each class

**Comment by ardian4** [comment]
@CuriousVini PR against develop branch (with this PR review fixes): https://github.com/data-integrations/google-cloud/pull/265

**Comment by CuriousVini** [comment]
@ardian4 Are above comments addressed?

**Comment by ardian4** [comment]
> @ardian4 Are above comments addressed? 

@CuriousVini yes, they are in this new PR #265 against develop branch

**Pull request closed without merging**